### ISSUE TITLE
new projection system

### ISSUE THREAD

**Issue by DavidBadura** [patchlevel/event-sourcing#295]
## new projection system requirement

* Projections should be able to be running asynchronously using a worker (#293)
* The position of a projection should be stored in a store so that the system always knows where a projection is and can continue from there
* It should be possible to prepare new projections when deploying, so that you don't have to wait long for the current projection when deploying (switching the application).
* New versions of a projection should be able to be built in parallel without changing the existing projection and affecting the running application.
* Old projections that are no longer needed (stale) should be able to be cleaned up.
* Faulty projections should be marked. This projection is skipped so that other projections can continue anyway.
* Faulty projections can either be restarted directly (from the current position) or be rebuilt from scratch.
* In general, you should be able to rebuild every projection from scratch.

## inspired blog post

* https://barryosull.com/blog/projection-building-blocks-what-you-ll-need-to-build-projections/
* https://barryosull.com/blog/managing-projectors-is-harder-than-you-think/

**Comment by robinlehrmann** [comment]
What do you want to change on the current projection implementation. I ask only out of interest. :)

**Comment by DavidBadura** [comment]
I updated the description ;-)
